android/src/main/java/com/guichaguri/trackplayer/service/metadata/MetadataManager.java

```diff
@@ -138,21 +138,31 @@ public void updateOptions(Bundle options) {
         }
 
         // Update the color
-        builder.setColor(options.getInt("color", NotificationCompat.COLOR_DEFAULT));
+        builder.setColor(getIntFromOptions(options, "color", NotificationCompat.COLOR_DEFAULT));
 
         // Update the icon
         builder.setSmallIcon(getIcon(options, "icon", R.drawable.play));
 
         // Update the jump interval
-        jumpInterval = options.getInt("jumpInterval", 15);
+        jumpInterval = getIntFromOptions(options, "jumpInterval", 15);
 
         // Update the rating type
-        ratingType = options.getInt("ratingType", RatingCompat.RATING_NONE);
+        ratingType = getIntFromOptions(options, "ratingType", RatingCompat.RATING_NONE);
         session.setRatingType(ratingType);
 
         updateNotification();
     }
 
+    private int getIntFromOptions(Bundle bundle, String key, int defaultValue) {
+        Object optionAsObject = bundle.get(key);
+        if (optionAsObject instanceof Integer) {
+            return bundle.getInt(key, defaultValue);
+        } else if (optionAsObject instanceof Double) {
+            return (int) bundle.getDouble(key, defaultValue);
+        }
+        return defaultValue;
+    }
```

Comment:
Can you move this method to `Utils` and rename it to `getInt`?

Reply:
updated